p2p/proto/state.proto

```diff
@@ -0,0 +1,41 @@
+syntax = "proto3";
+import "p2p/proto/common.proto";
+
+
+// optimized for flat storage, not through a trie (not sharing key prefixes)
+message ContractStoredValue {
+    Felt252 key = 1;
+    Felt252 value = 2;
+}
+
+message StateDiff
+{
+    // a bit more efficient than the state sync separation
+    message ContractDiff {
+        Address  address                    = 1;
+        optional Felt252 nonce              = 2;
+        optional Felt252 class_hash         = 3;  // can change for replace_class or new contract
+        repeated ContractStoredValue values = 4;
+    }
+
+    uint32   domain                      = 1;  // volition state domain
+    repeated ContractDiff contract_diffs = 2;
+}
+
+// is it better to separate the definition from the hashes? (will need to repeate the hashes
+// for the definitions stream)
+// or, make the definitions optional? maybe it is enough to know only that a class exists, not its definition
+// which may be fetched lazily later.
+message Class {
```

Comment:
This class message doesn't seem enough to support both the `Cairo0` and `Cairo1` classes and certainly not enough to support the relevant rpc calls. From what I can see it is missing `Abi`, `EntryPoint`s, and `Cairo1` fields. The following is how we have defined Cairo Classes in Juno:

```go
type Cairo0Class struct {
	Abi json.RawMessage
	Externals []EntryPoint
	L1Handlers []EntryPoint
	Constructors []EntryPoint
	Program string
}

type EntryPoint struct {
	Selector *felt.Felt
	Offset *felt.Felt
}

type Cairo1Class struct {
	Abi         string
	AbiHash     *felt.Felt
	EntryPoints struct {
		Constructor []SierraEntryPoint
		External    []SierraEntryPoint
		L1Handler   []SierraEntryPoint
	}
	Program         []*felt.Felt
	ProgramHash     *felt.Felt
	SemanticVersion string
	Compiled        json.RawMessage
}

type SierraEntryPoint struct {
	Index    uint64
	Selector *felt.Felt
}
```